src/Symfony/Bundle/WebServerBundle/WebServer.php

```diff
@@ -0,0 +1,209 @@
+        }
+
+        $pid = pcntl_fork();
+
+        if ($pid < 0) {
+            throw new \RuntimeException('Unable to start the server process.');
+        }
+
+        if ($pid > 0) {
+            return self::STARTED;
+        }
+
+        if (posix_setsid() < 0) {
+            throw new \RuntimeException('Unable to set the child process as session leader');
+        }
+
+        $process = $this->createServerProcess($router);
+        $process->disableOutput();
+        $process->start();
+
+        $lockFile = $this->getLockFile();
+        touch($lockFile);
+
+        if (!$process->isRunning()) {
+            unlink($lockFile);
+            throw new \RuntimeException('Unable to start the server process');
+        }
+
+        // stop the web server when the lock file is removed
+        while ($process->isRunning()) {
+            if (!file_exists($lockFile)) {
+                $process->stop();
+            }
+
+            sleep(1);
+        }
+
+        return self::STOPPED;
+    }
+
+    public function stop()
+    {
+        if (!file_exists($lockFile = $this->getLockFile())) {
+            throw new \RuntimeException(sprintf('No web server is listening on http://%s', $this->getAddress()));
+        }
+
+        unlink($lockFile);
+    }
+
+    public function isRunning()
+    {
+        if (false !== $fp = @fsockopen($this->hostname, $this->port, $errno, $errstr, 1)) {
+            fclose($fp);
+
+            return true;
+        }
+
+        if (file_exists($lockFile = $this->getLockFile())) {
+            unlink($lockFile);
+        }
+
+        return false;
+    }
+
+    /**
+     * @return Process The process
+     */
+    private function createServerProcess($router = null)
+    {
+        $finder = new PhpExecutableFinder();
+        if (false === $binary = $finder->find()) {
+            throw new \RuntimeException('Unable to find the PHP binary.');
+        }
+
+        $builder = new ProcessBuilder(array($binary, '-S', $this->getAddress(), $router));
+        $builder->setWorkingDirectory($this->documentRoot);
+        $builder->setTimeout(null);
+
+        return $builder->getProcess();
+    }
+
+    /**
+     * Determines the name of the lock file for a particular PHP web server process.
+     *
+     * @return string The filename
+     */
+    private function getLockFile()
+    {
+        return sys_get_temp_dir().'/'.strtr($this->getAddress(), '.:', '--').'.pid';
+    }
+
+    private function guessFrontController($documentRoot, $env)
+    {
+        foreach (array('app', 'index') as $prefix) {
+            $file = sprintf('%s_%s.php', $prefix, $env);
+            if (file_exists($documentRoot.'/'.$file)) {
+                return $file;
+            }
+
+            $file = sprintf('%s.php', $prefix);
+            if (file_exists($documentRoot.'/'.$file)) {
+                return $file;
+            }
+        }
+
+        return false;
```

Comment:
I prefer a ``string|null`` API than ``string|false``. The absence of value is null, not a boolean